File: Kata 6/Transform To Prime/Transform To Prime.cs
using System;
using System.Linq;

/*
 Find 
1- minimumNumber ({3,1,2}) ==> return (1)
sum(3,1,2) ->6
next prime num is 7

so return 7-6

*/
namespace TransformToPrime
{
  class Solution
  {   

	public static int MinimumNumber(int[] numbers, int n = 0)
    {
        if (IsPrime(numbers.Sum() + n)) return n;
        return MinimumNumber(numbers, ++n);
    }
    static bool IsPrime(int n)
    {
        if (n == 1) return false;
        if (n == 2) return true;
        return Enumerable.Range(2, (int)Math.Sqrt(n)).All(x => n % x > 0);
    }
    public static int MinimumNumber_(int[] numbers)
    {
		int sum=numbers.Sum();		
		int n=sum;
		while(Enumerable.Range(2, (int)Math.Sqrt(n)).Any(x => n % x ==0))
			n++;		
		return sum-n;		
	}	
      
  }
}

File: Kata 6/Transform To Prime/Transform To Prime.py
def minimum_number(numbers):
    sums=sum(numbers)
    while　True　:        
        if all (sums%i for i in range(2,int(sums**0.5)+1,1)):
            return sums-sum(numbers)
        sums=sums+1
 
 
def minimum_number(numbers):
    n=sum(numbers)  
    add=0
    while True:
        if is_prime(n+add):
            return add
        else:
            add += 1    
    return add

def is_prime(n):
    if n==1:
        return False
    elif n==2:
        return True
    else:
        return all ( [n%i for i in range(2,int(n**0.5)+1)])